py/dml/structure.py

```diff
@@ -2977,7 +3025,10 @@ def mkmethod(site, rbrace_site, location, parent_obj, name, inp_ast,
                       'Methods with (partially) const output/return '
                       + 'values are not yet supported.')
     if default_level:
-        name += "___default%d" % (default_level,)
+        if template:
+            name = f'templates__{template.name}__{name}'
+        else:
+            name += f'___default{default_level}'
```

Comment:
Figured this would make generated C easier to parse, especially for GCC or Coverity errors.